docs/applications/containers/troubleshooting-kubernetes/index.md

```diff
@@ -0,0 +1,204 @@
+
+### kubectl logs
+
+Use the [`logs` command](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#logs) to print logs collected by a pod:
+
+    kubectl logs mariadb-0
+
+-   You can use the `--selector` (`-l`) flag to print logs from all pods that match a selector:
+
+        kubectl logs -l app=ghost
+
+-   If a pod's container was killed and restarted, you can view the previous container's logs with the `--previous` or `-p` flag:
+
+        kubectl logs -p ghost-0
+
+### kubectl exec
+
+You can run arbitrary commands on a pod's container by passing them to kubectl's [`exec` command](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#exec):
+
+    kubectl exec mariadb-0 -- ps aux
+
+The full syntax for the command is:
+
+    kubectl exec ${POD_NAME} -c ${CONTAINER_NAME} -- ${CMD} ${ARG1} ${ARG2} ... ${ARGN}
+
+{{< note >}}
+The `-c` flag is optional, and is only needed when the specified pod is running more than one container.
+{{< /note >}}
+
+It is possible to run an interactive shell on an existing pod/container. Pass the `-it` flags to `exec` and run the shell:
+
+    kubectl exec -it mariadb-0 -- /bin/bash
+
+Enter `exit` to later leave this shell.
+
+### Viewing Master and Worker Logs
+
+If you have set up Kubernetes using a method that does not include [**systemd**](https://linode.com/docs/quick-answers/linux-essentials/what-is-systemd/), the location of logs on your master nodes are in most cases as follows:
+
+- `/var/log/kube-apiserver.log` - [API server](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/)
+- `/var/log/kube-scheduler.log` - [Scheduler](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/ )
+- `/var/log/kube-controller-manager.log` - [Replication controller manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)
+
+On your worker nodes:
+
+- `/var/log/kubelet.log` - [Kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)
+- `/var/log/kube-proxy.log` - [Kube proxy](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/)
+
+If you do have systemd implemented, then the above logs will not be accessible and you'll need to access kubelet logs using `journalctl` instead. Below is an example command for accessing kubelet logs through journalctl:
+
+    journalctl --unit kubelet
+
+In some cases, the information you find from kubelet may prompt you to look into individual container logs for more information. When using Docker as a container runtime, you can find information on all running containers with the `docker ps` command. For example, to find the container running your API server, enter the following:
+
+    docker ps | grep apiserver
+
+Then, using the container ID that appears, enter `docker logs` to view the logs for your API server:
+
+    docker logs CONTAINER_ID
```

Comment:
I am gonna put an item in our draft style guide to find a standard that we can agree on when referring to variable names, because I think we tend to use different styles in different places. I kind of prefer to use `${VARIABLE_NAME} lately because bash will substitute it. This isn't really a problem specific to this line or this guide, it's just reminded me that we should do that